clerk/keeper.go

```diff
@@ -156,6 +182,34 @@ func (k *Keeper) GetEventRecordList(ctx sdk.Context, page uint64, limit uint64)
 	return records, nil
 }
 
+// GetEventRecordListWithTime returns all records with params like fromTime and toTime
+func (k *Keeper) GetEventRecordListWithTime(ctx sdk.Context, fromTime time.Time, toTime time.Time) ([]types.EventRecord, error) {
+	store := ctx.KVStore(k.storeKey)
+
+	// create records
+	var records []types.EventRecord
+
+	// get range iterator
+	fromTimeBytes := sdk.FormatTimeBytes(fromTime)
+	toTimeBytes := sdk.FormatTimeBytes(toTime)
+	iterator := store.Iterator(append(StateRecordPrefixKeyWithTime, fromTimeBytes...), append(StateRecordPrefixKeyWithTime, toTimeBytes...))
```

Comment:
Once we have `EventRecordByTimeKey`, we can do this:

```suggestion
	iterator := store.Iterator(EventRecordByTimeKey(fromTime), EventRecordByTimeKey(toTime))
```